src/QsCompiler/Transformations/CallGraph.cs

```diff
@@ -54,6 +55,72 @@ internal static bool TryCombineTypeResolutionsForTarget(QsQualifiedName target,
             return success;
         }
 
+        /// <summary>
+        /// Reversed the dependencies of type parameters resolving to other type parameters in the given
+        /// dictionary to create a lookup whose keys are type parameters and whose values are all the type
+        /// parameters that can be updating by knowing the resolution of the lookup's associated key.
+        /// </summary>
+        private static ILookup<Tuple<QsQualifiedName, NonNullable<string>>, Tuple<QsQualifiedName, NonNullable<string>>> GetReplaceable(TypeParameterResolutions.Builder typeParamResolutions)
+        {
+             return typeParamResolutions
+                .Select(kvp => (kvp.Key, GetTypeParameters.Apply(kvp.Value)))   // Get any type parameters in the resolution type.
+                .SelectMany(tup => tup.Item2.Select(value => (tup.Key, value))) // For each type parameter found, match it to the dictionary key.
+                .ToLookup(                                                      // Reverse the keys and resulting type parameters to make the lookup.
+                    kvp => kvp.value,
+                    kvp => kvp.Key);
+        }
+
+        /// <summary>
+        /// Applies all the type parameter resolutions in a dictionary to the whole dictionary,
```

Comment:
```suggestion
        /// Adds all the type parameter resolutions in an independent dictionary to a combined one,
```

Reply:
I've tried to reword the description here. I want to make sure to reflect that the individual resolutions are being applied to the other resolutions in the same dictionary. Let me know if the new description is any clearer.

Reply:
The rewording is better than these small changes :)